x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/TransportRollupAction.java

```diff
@@ -109,137 +109,184 @@ protected void masterOperation(
         RollupAction.Request request,
         ClusterState state,
         ActionListener<AcknowledgedResponse> listener
-    ) throws IOException {
-        String originalIndexName = request.getSourceIndex();
-
-        final String rollupIndexName;
-        if (request.getRollupIndex() == null) {
-            rollupIndexName = "rollup-" + originalIndexName + "-" + UUIDs.randomBase64UUID(Randomness.get());
-        } else {
-            rollupIndexName = request.getRollupIndex();
+    ) {
+        String sourceIndexName = request.getSourceIndex();
+        IndexMetadata sourceIndexMetadata = state.getMetadata().index(sourceIndexName);
+        // Assert source index exists
+        if (sourceIndexMetadata == null) {
+            listener.onFailure(new IndexNotFoundException(sourceIndexName));
+            return;
+        }
+        // Assert source index is a time_series index
+        if (IndexSettings.MODE.get(sourceIndexMetadata.getSettings()) != IndexMode.TIME_SERIES) {
+            listener.onFailure(
+                new ElasticsearchException(
+                    "Rollup requires setting ["
+                        + IndexSettings.MODE.getKey()
+                        + "="
+                        + IndexMode.TIME_SERIES
+                        + "] for index ["
+                        + sourceIndexName
+                        + "]"
+                )
+            );
+            return;
+        }
+        // Assert source index is read-only
+        if (state.blocks().indexBlocked(ClusterBlockLevel.WRITE, sourceIndexName) == false) {
+            listener.onFailure(
+                new ElasticsearchException(
+                    "Rollup requires setting [" + IndexMetadata.SETTING_BLOCKS_WRITE + " = true] for index [" + sourceIndexName + "]"
+                )
+            );
```

Comment:
missing return statement?

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/85708/commits/cd23fd713ba6adbaa3be8c3987f27900b6a0b476